### ISSUE TITLE
Add a DiskArrays extension

### ISSUE THREAD

**Issue by rafaqz** [rafaqz/DimensionalData.jl#770]
As wrapping some kind of DiskArray is a major use case of DimensionalData we may as well move the diskarrays forwarding from Rasters/YAXArrays/PyramidScheme to an extension here, and then they will work in any other wrapper too.

This will mean we can do proper testing of broadcasting and iteration over wrapped DiskArrays here rather than in the extending packages.

For motivation see: https://github.com/meggart/DiskArrays.jl/issues/185

@felixcremer @meggart does that make sense to you?

**Comment by felixcremer** [comment]
Yes, I think it would make sense to streamline this and make the interaction clearer between Dimensions and DiskArrays.

**Comment by rafaqz** [comment]
We have this now

**Comment by asinghvi17** [comment]
I think we still don't forward a bunch of stuff (`isdisk` etc).  Should we do that?

**Comment by rafaqz** [comment]
Is disk was only added to DiskArrays last week ;)

But forward away, that's why I added it
